irods/test/data_obj_test.py

```diff
@@ -2164,64 +2172,56 @@ def update(self,n):
         # In this test, registering the instance type allows us to use the progressbar instance in the updatables list
         from irods.manager.data_object_manager import (register_update_type, unregister_update_type)
 
-        try:
-            # This is the ProgressBar from either of the following sources:
-            #    - https://pypi.org/project/progressbar/
-            #    - https://pypi.org/project/progressbar2/
-            progressbar_class = progressbar.ProgressBar
-            weakref.ref(progressbar_class(maxval = 10))
-        except TypeError:
-            class ProgressBar_wrapper:
-               def start(self,*x):
-                   return self.p.start(*x)
-               def update(self,*x):
-                   return self.p.update(*x)
-               @property
-               def currval(self):
-                   return self.p.currval
-               def __init__(self,*args,**kw):
-                   self.p = progressbar.ProgressBar(*args,**kw)
-
-            progressbar_class = ProgressBar_wrapper
+        # This is the ProgressBar from either of the following sources:
+        #    - https://pypi.org/project/progressbar/
+        #    - https://pypi.org/project/progressbar2/
+        # Wrapping the ProgressBar instances is useful in the case of both of the above libraries,
+        # due to their instances being callable - which confuses our methodology for managing registered objects - or,
+        # in the former case, being unusable as weak references.
+
+        class ProgressBar_wrapper:
+            def start(self,*x):
+                return self.pbar.start(*x)
+            def update(self,*x):
+                return self.pbar.update(*x)
+            @property
+            def value(self):
+                try: 
+                    return self.pbar.value
+                except AttributeError:
+                    return self.pbar.currval
+            def __init__(self,*args,**kw):
+                self.pbar = progressbar.ProgressBar(*args,**kw)
 
         def adapt_ProgressBar(pbar):
-            # Create a function closure containing a state variable and a lock object for the instance to be updated.
             total = [0]
             l = threading.Lock()
             pbar.start()
-
-            # Here's the actual updating callable that will be returned for the current pbar instance (also part of the closure)
-            # and then registered for update during the data transfer.
             def _update(n):
-                # total was made a list-of-single-integer for the benefit of Python2 clients.
-                # If total were a bare integer we'd declare it by "nonlocal total" in Python3 and it would just work.
                 with l:
-                    # this type of progress bar requires a cumulative sum as its argument to the update method.
                     total[0] += n
                     pbar.update(total[0])
-
             return _update
```

Comment:
Why were all these comments removed?

Reply:
I see... Yes, what happened is I removed them for my own clarity during test debugging and forgot to restore them in the final commit.

Reply:
Rectified.

Reply:
Makes sense. Looks like this can be resolved. Thanks